openapi/generator.go

```diff
@@ -878,10 +878,11 @@ func (g *Generator) newSchemaFromType(t reflect.Type) *SchemaOrRef {
 	}
 	dt := g.datatype(t)
 
-	// Check for "Nullable"
-	i, ok := reflect.New(t).Interface().(Nullable)
-	if ok {
-		nullable = i.Nullable()
+	if t.Implements(tofNullable) {
+		i, ok := reflect.New(t).Interface().(Nullable)
+		if ok {
+			nullable = i.Nullable()
```

Comment:
This overrides the assignation made line 877, if the type is a pointer.
This should be taken into consideration in priority, even if the interface's method return false next.
This should be covered by a test with a pointer type to a `Nullable` interface to ensure that pointers are always marked as nullable in the spec.